aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -388,10 +388,20 @@ struct NormTwoOps {
 #endif
 };
 
+template <typename scalar_t>
+C10_DEVICE bool isnan(scalar_t b) {
+    return at::_isnan(b);
+}
+
+template <typename scalar_t>
+C10_DEVICE bool isnan(c10::complex<scalar_t> b) {
```

Comment:
FYI: potential code duplication with: https://github.com/pytorch/pytorch/pull/93153

Reply:
Thanks for pointing it out, @Skylion007! I wasn't aware of it. I've removed this function overload.

Also, I think there we'll need to dispatch from both `__host__` and `__device__`, because here (in NanSumOps struct), the functions are defined in C10_DEVICE. (changing here might cause performance regression). I've updated that part, adding C10_HOST_DEVICE. Please correct me if I'm wrong or if there's a better way.

Thanks!

Reply:
You can follow the discussion in the original PR: https://github.com/pytorch/pytorch/pull/90847 (and why the CUDA code wasn't included due to a mysterious 30 minute build time regression). It might be it makes more sense to use another method like the aten version of is_nan if it already has that dispatch?

Reply:
> It might be it makes more sense to use another method like the aten version of is_nan if it already has that dispatch?

Thanks!! I stumbled a bit and found an aten version for `isnan`. I think now to even don't need the `c10/...` version.

Reply:
> It might be it makes more sense to use another method like the aten version of is_nan if it already has that dispatch?

Indeed, thanks for mentioning it! I looked through the code and found aten version already available for `isnan`. I have updated that for using it on the `__device__` side. Also, We might not need the `c10/...` version now. What do you think?